### PR TITLE
feat: finalize pipeline with inference & tail question generation

### PR DESCRIPTION
**Pull request by pega501**
#### Related Issue(s):

N/A (initial pipeline completion milestone)

#### PR Description:

This PR finalizes the XGBoost-based training and generation pipeline, integrating training, inference, tail question generation, and testing.
It includes class-weighted training for balanced regression, a modularized inference framework, and full-pipeline validation with timing analysis.

##### Changes Included:

- Finalized XGBoost training pipeline
- Added inference logic
- Implemented few-shot tail question generation
  - Replaced the previous ReAct method with few-shot prompting due to excessive response time
- Added full-pipeline testing with execution time measurement
 - Current bottlenecks: **acoustic feature generation** and **tail question generation**

##### Screenshots (if UI changes were made):

N/A

##### Notes for Reviewer:

- Check that the end-to-end test (test_pipeline_full) runs successfully with both synthetic and real data.
- Ensure requirements.txt is consistent with your local environment before deployment.
- Ensure API keys are in .env file.

---

#### Reviewer Checklist:

- [ ] Code is written in clean, maintainable, and idiomatic form.
- [ ] Automated test coverage is adequate.
- [ ] All existing tests pass.
- [ ] Manual testing has been performed to ensure the PR works as
      expected.
- [ ] Code review comments have been addressed or clarified.

---

#### Additional Comments:

Future work: Reduce total runtime to enable tail question generation within a reasonable time frame, focusing on optimizing feature extraction and few-shot generation efficiency.

### CODE REVIEW

File: research/train.py [hunk 1]
```diff
@@ -171,12 +182,10 @@ def main():
     mae = mean_absolute_error(y_valid, y_pred)
 
     # 회귀를 등급으로 변환: 반올림
-    y_pred_rounded = np.clip(np.round(y_pred), 1, 8)  # 1~8 범위로 클립
-    accuracy = accuracy_score(y_valid, y_pred_rounded)
+    y_pred_rounded = np.clip(np.round(y_pred), 1, 8)
 
     print(f"검증 RMSE: {rmse:.4f}")
     print(f"검증 MAE : {mae:.4f}")
-    print(f"검증 Accuracy(반올림 기준): {accuracy:.2%}")
```
[the remark this answers is not in the document]
Comment (pega501) [reply research/train.py]: Validation Set이 작아서 세분화된 등급에서는 accuracy 변동이 커서, 보다 안정적인 평가를 위해 A/B/C/D로 단순화한 accuracy를 산출했습니다.
